unit_tests/test_matching_algorithms/test_preferred_partners_matching_algorithm.py

```diff
@@ -0,0 +1,126 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import uuid
+from typing import Optional
+
+from d3a_interface.data_classes import Offer, BidOfferMatch, Bid
+from pendulum import DateTime
+
+from myco_api_client.matching_algorithms.preferred_partners_algorithm import (
+    PreferredPartnersMatchingAlgorithm)
+
+
+class TestPreferredPartnersMatchingAlgorithm:
+
+    @staticmethod
+    def offer_factory(additional_data: dict = None):
```

Comment:
Nit: this can be a fixture (and you can pass parameters with `pytest.mark.parametrize`)

Reply:
I personally don't see a need for parametrized fixtures here, the factory method is more than enough, and It gives us more control inside the test functions (in some methods I create more orders `offers.append(self.offer_factory({..}))
If you believe there are an extra benefit or styling improvement in using parametrize+fixtures, let me know

Reply:
You can use fixtures to achieve the same result (creating more orders). In general, pytest fixtures are not necessary: the same things can be done by implementing custom-made functions each time. The advantage of using pytest is that this has been already implemented for us and that if we stick to a unique style (e.g. using pytest fixtures) the codebase becomes more coherent. Anyway, this might be a personal opinion, that's why I posted this as a "nit".

Reply:
Thanks for giving it a try! I trust your judgement 👍